bodhi/server/services/releases.py

```diff
@@ -324,6 +324,17 @@ def save_release(request):
             log.info("Editing release: %s" % edited)
             r = request.db.query(Release).filter(Release.name == edited).one()
             for k, v in data.items():
+                # We have to change updates status to obsolete
+                # if state of release changes to archived
+                if k == "state" and v == ReleaseState.archived and \
+                        r.state != ReleaseState.archived:
+                    updates = request.db.query(Update).filter(Update.release_id == r.id).filter(
+                        Update.status.notin_(
+                            [UpdateStatus.obsolete, UpdateStatus.stable, UpdateStatus.unpushed]
+                        )
+                    ).all()
+                    for u in updates:
+                        u.status = UpdateStatus.obsolete
```

Comment:
CC @mohanboddu

Reply:
To clarify a bit, by untag, I mean this function:

https://github.com/fedora-infra/bodhi/blob/3.11.2/bodhi/server/models.py#L3198-L3215

It should only remove tags that Bodhi knows about, which are tags returned from this function:

https://github.com/fedora-infra/bodhi/blob/3.11.2/bodhi/server/models.py#L929-L953

Reply:
The only thing is, if we untag a build, there is a chance that it might get untagged from all tags, if and when it happens the build might get garbage collected. Even then I dont think there would be problem as we garbage collect lot of other builds.

So, I think it should be fine. But, I would like to get other's opinions as well.

Reply:
OK, now that I've had time to think about this some more, I'm convinced that we should not use the obsoletion function and the way you've done it here is good.

One thing I think we should do, however, is add a comment on each update when we change it to obsoleted state. Something like this:

```
u.comment('This update is marked obsolete because the {} release is archived.'.format(u.release.name), agent='bodhi')
```